scripts/exglobal_atmos_products.sh

```diff
@@ -29,254 +29,245 @@ fi
 # Determine grids once and save them as a string and an array for processing
 grid_string="0p25"
 if [[ "${PGBS:-}" == "YES" ]]; then
-  grid_string="${grid_string}:0p50:1p00"
+    grid_string="${grid_string}:0p50:1p00"
 else
-  echo "INFO: Supplemental product generation is disabled for fhr = ${fhr3}"
-  PGBS="NO"  # Can't generate supplemental products if PGBS is not YES
+    echo "INFO: Supplemental product generation is disabled for fhr = ${fhr3}"
+    PGBS="NO" # Can't generate supplemental products if PGBS is not YES
 fi
 # Also transform the ${grid_string} into an array for processing
 IFS=':' read -ra grids <<< "${grid_string}"
 
 # Files needed by ${USHgfs}/interp_atmos_master.sh
 MASTER_FILE="${COMIN_ATMOS_MASTER}/${PREFIX}master.${fhr3}.grib2"
 
-nset=1
-while [[ ${nset} -le ${downset:-1} ]]; do
+for ((nset = 1; nset <= downset; nset++)); do
 
-  echo "INFO: Begin processing nset = ${nset}"
+    echo "INFO: Begin processing nset = ${nset}"
 
-  # Each set represents a group of files
-  if [[ ${nset} == 1 ]]; then
-    grp="a"
-  elif [[ ${nset} == 2 ]]; then
-    grp="b"
-  fi
-
-  # Get inventory from ${MASTER_FILE} that matches patterns from ${paramlist}
-  # Extract this inventory from ${MASTER_FILE} into a smaller tmpfile based on paramlist
-
-  tmpfile="tmpfile${grp}_${fhr3}"
-  paramlist="paramlist${grp}"
-  parmfile="${!paramlist}"
-
-  # shellcheck disable=SC2312
-  ${WGRIB2} "${MASTER_FILE}" | grep -F -f "${parmfile}" | ${WGRIB2} -i -grib "${tmpfile}" "${MASTER_FILE}" && true
-  export err=$?
-  if [[ ${err} -ne 0 ]]; then
-    err_exit "wgrib2 failed to create intermediate grib2 file from '${MASTER_FILE}' using '${parmfile}'"
-  fi
-
-  # Number of processors available to process $nset
-  nproc=${ntasks}
-
-  # shellcheck disable=SC2312
-  ncount=$(${WGRIB2} "${tmpfile}" | wc -l)
-  if [[ ${nproc} -gt ${ncount} ]]; then
-    echo "WARNING: Total no. of available processors '${nproc}' exceeds no. of records '${ncount}' in ${tmpfile}"
-    echo "WARNING: Reduce nproc to ${ncount} (or less) to not waste resources"
-  fi
-  inv=$(( ncount / nproc ))
-  rm -f "${DATA}/cmdfile"
-
-  last=0
-  iproc=1
-  while [[ ${iproc} -le ${nproc} ]]; do
-    first=$((last + 1))
-    last=$((last + inv))
-    if [[ ${last} -gt ${ncount} ]]; then
-       last=${ncount}
+    # Each set represents a group of files
+    if [[ "${nset}" == 1 ]]; then
+        grp="a" # TODO: this should be "a" when we eventually rename the pressure grib2 files per EE2 convention
```

Comment:
This work has been completed
```suggestion
        grp="a"
```

Reply:
Fixed